### ISSUE TITLE
Operator displays successful upgrade but is not upgraded; too many characters in name 

### ISSUE THREAD

**Issue by jcpowermac** [eclipse-che/che#19506]
### Describe the bug

Upgrading from 7.27.2 to 7.28 fails from operator logs:

```
time="2021-04-05T21:37:43Z" level=error msg="CheCluster.org.eclipse.che \"eclipse-che\" is invalid: metadata.finalizers: Invalid value: \"eclipse-che-che-che-operator.clusterrolebinding.finalizers.che.eclipse.org\": name part must be no more than 63 characters"
```

### Che version

 - [x] latest
 - [ ] nightly
 - [ ] other: please specify

### Runtime
   - [ ] kubernetes (include output of `kubectl version`)
   - [x] Openshift (include output of `oc version`)

``` 
$ oc version
Client Version: 4.6.0-0.nightly-2020-12-21-110104
Server Version: 4.7.0
Kubernetes Version: v1.20.0+bd9e442
```
   - [ ] minikube (include output of `minikube version` and `kubectl version`)
   - [ ] minishift (include output of `minishift version` and `oc version`)
   - [ ] docker-desktop + K8S (include output of `docker version` and `kubectl version`)
   - [ ] other: (please specify)

### Installation method
   - [ ] chectl
     * provide a full command that was used to deploy Eclipse Che (including the output)
     * provide an output of `chectl version` command
   - [x] OperatorHub
   - [ ] I don't know

### Environment
   - [ ] my computer
       - [ ] Windows
       - [ ] Linux
       - [ ] macOS
   - [ ] Cloud
       - [x] Amazon
       - [ ] Azure
       - [ ] GCE
       - [] other (please specify)
   - [ ] Dev Sandbox (workspaces.openshift.com)
   - [ ] other: please specify

**Comment by jcpowermac** [comment]
Operator log:
[log.txt](https://github.com/eclipse/che/files/6260826/log.txt)

**Comment by jcpowermac** [comment]
![image](https://user-images.githubusercontent.com/5401402/113631543-9c9f8600-9637-11eb-9ce4-73ed72d2da14.png)

![image](https://user-images.githubusercontent.com/5401402/113631634-bb9e1800-9637-11eb-9b6b-69402bc7223b.png)

**Comment by tolusha** [comment]
@jcpowermac 
Thank you for reporting. We will provide a fix.

**Comment by tolusha** [comment]
Fixed in 7.28.2, 7.29.0
